cg/meta/upload/scout/scout_config_builder.py

```diff
@@ -37,14 +46,95 @@ def add_common_info_to_load_config(self) -> None:
+                "status": link_obj.status,
+            }
+            for link_obj in family_obj.links
+        ]
+        svg_path: Path = self.madeline_api.run(family_id=family_obj.name, samples=samples)
+        return svg_path
+
+    @staticmethod
+    def is_multi_sample_case(load_config: ScoutLoadConfig) -> bool:
+        return len(load_config.samples) > 1
+
+    @staticmethod
+    def is_family_case(load_config: ScoutLoadConfig) -> bool:
+        """Check if there are any linked individuals in a case."""
+        for sample in load_config.samples:
+            if sample.mother and sample.mother != "0":
+                return True
+
+            if sample.father and sample.father != "0":
+                return True
+        return False
+
+    def include_pedigree_picture(self) -> ScoutLoadConfig:
+        if self.is_multi_sample_case(load_config=self.load_config):
+            if self.is_family_case(load_config=self.load_config):
+                svg_path: Path = self.run_madeline(self.analysis_obj.case)
+                self.load_config.madeline = str(svg_path)
+            else:
+                LOG.info("family of unconnected samples - skip pedigree graph")
+        else:
+            LOG.info("family of 1 sample - skip pedigree graph")
+        return self.load_config
+
+    @staticmethod
+    def remove_chromosome_substring(file_path: str | None) -> str | None:
+        """Remove a file's suffix and identifying integer or X/Y
+        Example:
+        `/some/path/gatkcomb_rhocall_vt_af_chromograph_sites_X.png` becomes
+        `/some/path/gatkcomb_rhocall_vt_af_chromograph_sites_`"""
+        if file_path is None:
+            return file_path
+        return re.split(r"(?:\d+|[XY])\.png$", file_path)[0]
+
+    def get_sample_information(self, load_config: ScoutLoadConfig) -> list[ScoutIndividual]:
+        LOG.info("Building samples")
+        db_sample: CaseSample
+
+        for db_sample in self.analysis_obj.case.links:
+            load_config.samples.append(self.build_config_sample(case_sample=db_sample))
+        return load_config
+
+    def build_config_sample(self, case_sample: CaseSample) -> ScoutIndividual:
+        """Build a sample with rnafusion specific information."""
+        if Workflow.BALSAMIC in self.analysis_obj.workflow:
+            config_sample = ScoutCancerIndividual()
+        elif self.analysis_obj.workflow == Workflow.RAREDISEASE:
+            config_sample = ScoutRarediseaseIndividual()
+        elif self.analysis_obj.workflow == Workflow.MIP_DNA:
+            config_sample = ScoutMipIndividual()
+        elif self.analysis_obj.workflow == Workflow.RNAFUSION:
+            config_sample = ScoutIndividual()
+        config_sample = self.add_common_sample_info(
+            config_sample=config_sample, case_sample=case_sample
+        )
+        config_sample = self.add_common_sample_files(
+            config_sample=config_sample, case_sample=case_sample
+        )
+        return config_sample
```

Comment:
Very nice gathering of logic! It does seem like Balsamic has its own implementation of `build_config_sample` though. Does it not fit into this logic?

Reply:
Indeed, we overwrite the phenotype from case_sample.status with 
```
       if BalsamicAnalysisAPI.get_sample_type(sample_obj=case_sample.sample) == SampleType.TUMOR:
            config_sample.phenotype = PhenotypeStatus.AFFECTED.value
        else:
            config_sample.phenotype = PhenotypeStatus.UNAFFECTED.value
```

Reply:
same with the analysis type

Reply:
I see. I don't think we should go in there and muck around, but the `if Workflow.BALSAMIC` seems to suggest that the Balsamic logic goes through this code, while this is not true. I say remove the first If-statement